### DISCUSSION TITLE
insuficient data in dataframe

### DISCUSSION THREAD

**Discussion by PhilippeDo** [cuebook/CueObserve#167] [Q&A] [answered]
Hi,

How many records must we have in dataset to run the anomaly detection ?
Currently I have 136 records and I still have the "insuficient data in dataframe" message.

Thanks
BR

**Comment by sachinkbansal** [comment, marked as answer]
@PhilippeDo Each dataframe must have at least 20 data points **after aggregation**.

See https://cueobserve.cuebook.ai/anomaly-detection#generate-timeseries-forecast for how CueObserve generates dataframes from dataset SQL.

**Reply by PhilippeDo** [reply]
OK thank you @sachinkbansal, this error is now fixed, I was wrong regarding the number of records in my database, it was less than 20 !!!
